src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala

```diff
@@ -111,62 +105,12 @@ abstract class ExtensionMethods extends Transform with TypingTransformers {
         if (unboxed.isDerivedValueClass) checkNonCyclic(pos, seen + clazz, unboxed)
       }
 
-   /** We will need to clone the info of the original method (which obtains clones
-    *  of the method type parameters), clone the type parameters of the value class,
-    *  and create a new polymethod with the union of all those type parameters, with
-    *  their infos adjusted to be consistent with their new home. Example:
-    *
-    *    class Foo[+A <: AnyRef](val xs: List[A]) extends AnyVal {
-    *      def baz[B >: A](x: B): List[B] = x :: xs
-    *      // baz has to be transformed into this extension method, where
-    *      // A is cloned from class Foo and  B is cloned from method baz:
-    *      // def extension\$baz[B >: A <: Any, A >: Nothing <: AnyRef](\$this: Foo[A])(x: B): List[B]
-    *    }
-    *
-    *  TODO: factor out the logic for consolidating type parameters from a class
-    *  and a method for re-use elsewhere, because nobody will get this right without
-    *  some higher level facilities.
-    */
-    def extensionMethInfo(extensionMeth: Symbol, origInfo: Type, clazz: Symbol): Type = {
```

Comment:
I moved this to `internal.Types` as I was not sure how to transform symbols coming from tasty as ExtensionMethods only works on trees, and there is no flag to detect if a symbol of a tree is from tasty